src/problem.ts

```diff
@@ -1 +1,61 @@
-export class Problem {};
\ No newline at end of file
+import { CONTENT_TYPE, Problem } from "types";
+export class AsyncApiProblem extends Error implements Problem {
+
+    public "Content-Type" = CONTENT_TYPE;
+    public type: string;
+    public title: string;
+    public errorUrlPrefix: string;
+    public status: number;
+    public instance?: string;
+    public skipFields?: string[];
+    public detail?: string;
+    public parsedJSON?: object | undefined;
+    public validationErrors?: object[] | undefined;
+    public location?: object[] | undefined;
+    public refs?: object[] | undefined;
+    [key: string]: any;
+
+
+    constructor(problem: Problem) {
+        super(problem.detail || problem.title);
+        this.errorUrlPrefix = problem.errorUrlPrefix;
+        this.instance = problem.instance;
+        this.detail = problem.detail;
+        this.type =
+            problem
+                .type
+                .startsWith(problem.errorUrlPrefix)
+                ? problem.type
+                : `${problem.errorUrlPrefix}${problem.type}`;
+
+        this.title = problem.title;
+        this.status = problem.status;
+        this.skipFields = problem.skipFields || [];
+        this.parsedJSON = problem.parsedJSON;
+        this.validationErrors = problem.validationErrors
+        this.location = problem.location;
+        this.refs = problem.refs;
+
+        // initiate miscalleneous keys
+        for (let key in problem) {
+            if (this.skipFields.includes(key))
+                continue;
+            this[key] = problem[key];
+        }
```

Comment:
I think that we should move that logic to separate static function. We should have possibility to recreate from problem another problem with skipped props and recreate problem only with passed props like this:

```ts
static copy(problem: Problem, mode: 'skipProps' | 'leaveProps', props: string[]): Problem;
```